framework/src/modules/interoperability/mainchain/commands/cc_update.ts

```diff
@@ -12,18 +12,278 @@
+		const certificateValidity = checkCertificateValidity(partnerChainAccount, txParams.certificate);
+
+		if (certificateValidity.error) {
+			return certificateValidity;
+		}
+
+		// If params contains a non-empty activeValidatorsUpdate
+		const activeValidatorsValidity = checkActiveValidatorsUpdate(txParams);
+		if (activeValidatorsValidity.error) {
+			return activeValidatorsValidity;
+		}
+
+		const partnerChannelStore = context.getStore(transaction.moduleID, STORE_PREFIX_CHANNEL_DATA);
+		const partnerChannelData = await partnerChannelStore.getWithSchema<ChannelData>(
+			partnerChainIDBuffer,
+			channelSchema,
+		);
+		// Section: InboxUpdate Validity
+		const inboxUpdateValidity = checkInboxUpdateValidity(txParams, partnerChannelData);
+		if (inboxUpdateValidity.error) {
+			return inboxUpdateValidity;
+		}
+
+		return {
+			status: VerifyStatus.OK,
+		};
+	}
+
+	public async execute(
+		context: CommandExecuteContext<CrossChainUpdateTransactionParams>,
+	): Promise<void> {
+		const { header, params: txParams } = context;
+		const chainIDBuffer = getIDAsKeyForStore(txParams.sendingChainID);
+		const partnerChainStore = context.getStore(this.moduleID, STORE_PREFIX_CHAIN_DATA);
+		const partnerChainAccount = await partnerChainStore.getWithSchema<ChainAccount>(
+			chainIDBuffer,
+			chainAccountSchema,
+		);
+
+		const decodedCertificate = codec.decode<Certificate>(certificateSchema, txParams.certificate);
+
+		// if the CCU also contains a non-empty inboxUpdate, check the validity of certificate with liveness check
+		checkValidCertificateLiveness(txParams, header, decodedCertificate);
+
+		const partnerValidatorStore = context.getStore(this.moduleID, STORE_PREFIX_CHAIN_VALIDATORS);
+		const partnerValidators = await partnerValidatorStore.getWithSchema<ChainValidators>(
+			chainIDBuffer,
+			chainValidatorsSchema,
+		);
+
+		// Certificate and Validators Update Validity
+		checkCertificateTimestampAndSignature(
+			txParams,
+			partnerValidators,
+			partnerChainAccount,
+			decodedCertificate,
+			header,
+		);
+
+		// If params contains a non-empty activeValidatorsUpdate
+		checkValidatorsHashWithCertificate(txParams, decodedCertificate, partnerValidators);
+
+		// CCM execution
+		const beforeSendContext = createCCMsgBeforeSendContext({
+			feeAddress: context.transaction.senderAddress,
+			eventQueue: context.eventQueue,
+			getAPIContext: context.getAPIContext,
+			logger: context.logger,
+			networkIdentifier: context.networkIdentifier,
+			getStore: context.getStore,
+		});
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore);
+		let decodedCCMs;
+		try {
+			decodedCCMs = txParams.inboxUpdate.crossChainMessages.map(ccm => ({
+				serialized: ccm,
+				deserialized: codec.decode<CCMsg>(ccmSchema, ccm),
+			}));
+		} catch (err) {
+			await interoperabilityStore.terminateChainInternal(
+				txParams.sendingChainID,
+				beforeSendContext,
+			);
+
+			throw err;
```

Comment:
Just to be sure, do we really want to throw and exit here for schema errors since we don't throw but continue for the same at L241 ?

Reply:
Related to above comments, Clarifying with research